packages/kmod-5.10-nvidia/disable-gsp.service.in

```diff
@@ -0,0 +1,23 @@
+[Unit]
+Description=Disable GSP for selected instance types
+RequiresMountsFor=/etc/modprobe.d PREFIX/share/factory/etc/modprobe.d
+After=network-online.target
+Before=load-kernel-modules.service
+# Rerunning this service after the system is fully loaded will accomplish
+# nothing except write to a file that will never be used (after boot).
+# Refuse manual intervention.
+RefuseManualStart=true
+RefuseManualStop=true
+
+[Service]
+Type=oneshot
+ExecCondition=/usr/bin/shibaken does-value-start-with instance-type \
+  --value g4dn. --value g5. --value g5g.
+ExecStart=-/usr/bin/cp \
+  PREFIX/share/factory/etc/modprobe.d/disable-gsp.conf \
+  /etc/modprobe.d/disable-gsp.conf
```

Comment:
My reasoning:

- If we do not ignore the exit code and cp fails, that cascades into later units and may prevent boot.
- If we do ignore the exit code and cp fails, we boot as we would today. We have not disabled GSP firmware download, but the kmods load and boot proceeds.

I think that "boot proceeds" is the wrong outcome. In the case of intermittent failures in the cp command, having boot proceed but not do the expected "disable GSP firmware download" leaves the customer with an instance that does not work as expected. Better to have the dead instance, since orchestrators know how to deal with that scenario.